pennylane/transforms/diagonalize_measurements.py

```diff
@@ -60,6 +60,12 @@ def diagonalize_measurements(tape, supported_base_obs=_default_supported_obs, to
         circuit measurements, the :func:`split_non_commuting <pennylane.transforms.split_non_commuting>` transform
         can be applied.
 
+    .. note::
+        This transform will diagonalize what it can, i.e., `qml.X`, `qml.Y`, `qml.Z`, `qml.Hadamard`,
+        `qml.Identity`, or a linear combination of them. Any other observable will be skipped and
+        left undiagonalized. In the context of preprocessing a quantum circuit for execution on a
+        device, observable validation should be performed after this transform to ensure that all
+        undiagonalized observables are supported by the device.
```

Comment:
I would say something very similar to what you wrote in the PR description here! Something like this:

> The `diagonalize_measurements` transform assumes that `split_non_commuting` has already been applied, and will diagonalize measurements that are `X`, `Y`, `Z`, `Hadamard`, `Identity`, or any linear combination of the five. Any observables that are not recognized by `diagonalize_measurements` will not raise an error, deferring to the device's validation for supported measurements later on.

> Lastly, if `diagonalize_measurements` produces additional gates that the device does not support, the `decompose` transform will be applied to ensure that relevant additional gates are decomposed to those that the device supports.

Reply:
There is a note above this one that mentions `split_non_commuting`.

Reply:
Updated.